ArduCopter/mode_auto.cpp

```diff
@@ -408,6 +414,64 @@ void ModeAuto::circle_start()
     }
 }
 
+// returns true when we have completed doing loiter turns
+// false if we still have remaining tasks
+// also takes us to next states after we complete the current state task 
+bool Mode::_Circling::is_completed()
+{
+    switch (_state) {
+    case _Circling::State::MOVING_TO_EDGE:
+        if (copter.wp_nav->reached_wp_destination()) {
+            circling_start();
+        }
+        break;
+    case _Circling::State::CIRCLING:
+        // check if we have completed circling
+        if (fabsf(copter.circle_nav->get_angle_total()/M_2PI) >= _total_turns) {
+            // stop at circle edge where we started
+            copter.wp_nav->set_wp_destination_loc(circle_edge);
+            return true;
+        }
+        break;
+    }
+
+    return false;
+}
+
+float Mode::_Circling::get_distance_to_destination() const
+{
+    switch (_state) {
+    case _Circling::State::CIRCLING:
```

Comment:
```suggestion
    case State::CIRCLING:
```

similarly everywhere else you're already within the correct namespace.